stablehlo/reference/Ops.cpp

```diff
@@ -372,6 +403,56 @@ Tensor evalSliceOp(const Tensor &operand, Index startIndices, Sizes strides,
   return result;
 }
 
+SmallVector<Tensor> evalSortOp(ArrayRef<Tensor> inputs, Axis dimension,
+                               bool isStable, Region &comparator,
+                               Scope &scope) {
+  SmallVector<Tensor> runtimeResults;
+  for (auto input : inputs) runtimeResults.push_back(Tensor(input.getType()));
+  if (dimension < 0) dimension += inputs[0].getRank();
+  SmallVector<int64_t> indices(inputs.front().getShape()[dimension]);
+  auto inputShape = inputs.front().getShape();
+  inputShape[dimension] = 1;
+  auto inputSizes = Sizes(inputShape);
+  for (auto resultIt = IndexSpaceIterator(inputSizes, Index(inputShape.size()));
+       resultIt != IndexSpaceIterator(inputSizes, std::nullopt); ++resultIt) {
+    std::iota(indices.begin(), indices.end(), 0);
+    auto compare = [&](int64_t lhs, int64_t rhs) {
+      SmallVector<Tensor> args;
+      auto lhsIndex = *resultIt;
+      auto rhsIndex = *resultIt;
+      lhsIndex[dimension] = lhs;
+      rhsIndex[dimension] = rhs;
+      for (auto input : inputs) {
+        auto type =
+            comparator.front().getArgumentTypes()[0].dyn_cast<TensorType>();
```

Comment:
The arg type of an inputs[i] is not always equal to the type of the first bock argument but depends on which input[i] we are referring to. Note that the argument type for inputs[i] and inputs[i+1] could be different.

In this case type should be equal to something like  `comparator.front().getArgumentTypes()[2*input_index].dyn_cast<TensorType>();`